dashboard/lib/projects_list.rb

```diff
@@ -116,6 +117,34 @@ def fetch_featured_published_projects
       return featured_published_projects
     end
 
+    # Retrieve a class set of libraries for a specified class section
+    # @param section The section that has all students whose libraries should
+    #                be returned.
+    # @return [Hash<Array<Hash>>] A hash of lists of published libraries.
+    def fetch_section_libraries(section)
+      project_types = PUBLISHED_PROJECT_TYPE_GROUPS[:library]
+      section_students = section.students
+      [].tap do |projects_list_data|
+        student_storage_ids = PEGASUS_DB[:user_storage_ids].
+          where(user_id: section_students.pluck(:id)).
+          select_hash(:user_id, :id)
+        section_students.each do |student|
+          next unless student_storage_id = student_storage_ids[student.id]
+          PEGASUS_DB[:storage_apps].
+            where(storage_id: student_storage_id, state: 'active').
```

Comment:
This is going to generate more queries than you need. Instead, you can do it like this:
```
# first, get the list of student ids
student_storage_ids = section_students.map do |student|
  student_storage_ids[student.id]
end.compact

# then, make one query for any project by any student in your list
PEGASUS_DB[:storage_apps].
  where(storage_id: student_storage_ids, state: 'active').
```